cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -456,7 +463,11 @@ public void updateAppSpec(ApplicationId appId, ApplicationSpecification spec) th
       LOG.trace("Application {} exists in mds with specification {}", appId, GSON.toJson(existing));
     }
     ApplicationMeta updated = new ApplicationMeta(existing.getId(), spec);
-    writeApplicationSerialized(appId.getNamespace(), appId.getApplication(), appId.getVersion(), GSON.toJson(updated));
+    // Generate the creation time for the version of the app.
+    Long created = System.currentTimeMillis();
```

Comment:
at the Store layer, it's better to take the created time as an argument instead of generating it here. This allows potentially allows the caller to use the same timestamp, but also lets you set the exact value in unit tests.